core/deployment/src/main/java/io/quarkus/deployment/console/AeshConsole.java

```diff
@@ -56,7 +56,7 @@ public class AeshConsole extends QuarkusConsole {
      * Because Aesh can log deadlocks are possible on Windows if a write fails, unless care
      * is taken.
      */
-    private final LinkedBlockingDeque<String> writeQueue = new LinkedBlockingDeque<>();
+    private final ConcurrentLinkedDeque<String> writeQueue = new ConcurrentLinkedDeque<>();
```

Comment:
I saw your comment in the description but I'm curious how `AeshConsole` interacts with Reactive Messaging?

Reply:
JNI calls that can pin the carrier thread?

Reply:
I don't know if it's a plain old deadlock that happens just more frequently with VT (because you have more) or if it's a pinning issue.

Reply:
@anavarr @ozangunalp were you able to find what kind of deadlock it is? 

@gsmet be prepared... VTs are stressing parts of our code in some unexpected ways, but that won't be the only place where our synchronization is going to lead to issues (issues that would have eventually popped up even with platform threads - just not often enough to be a problem).

Reply:
@gsmet it doesn't. However, it causes a deadlock, in dev mode of course, when multiple virtual threads write sys.out concurrently. We haven't seen that issue when logging.

Reply:
@jponge I am not sure if this is a pinning issue, but fairly sure that it isn't related to JNI. The issue is reproducible and I've the thread dumps if anyone is interested.

Reply:
Pinning is definitely involved - so you wait for a lock and this wait pins meaning that nothing can make progress...